### DISCUSSION TITLE
Where to see device on the map?

### DISCUSSION THREAD

**Discussion by klehnst** [Skipperro/pajgps-homeassistant#4] [General]
I have PAJ Vehicle Finder 4G 1.0 v1.
For me only 2 entities are displayed, once speed in kmh and under diagnosis (present/absent) from the PAJ GPS,
Unfortunately the coordinates cannot be seen.
![Screenshot 2024-02-25 095328](https://github.com/Skipperro/pajgps-homeassistant/assets/46057062/7f27475e-1ad4-4d81-a06c-7343f8058c79)

**Comment by skipperro** [comment]
Yes, this seems to be right for devices without battery:
![image](https://github.com/Skipperro/pajgps-homeassistant/assets/20705647/4303a590-2c26-47ca-8adc-7300ce28d2b8)
One is for speed and another will tell you in which area the GPS device is. Away/Abwesend means - outside any defined area.

The coordinates are a part of the device-tracker attributes:
![image](https://github.com/Skipperro/pajgps-homeassistant/assets/20705647/bd719c97-7a60-4643-a90e-a81c512a3786)

So you can use this device and select it on the list of devices to track on the map. In my case I created a "person" for my car and attached a tracker device to it:
![image](https://github.com/Skipperro/pajgps-homeassistant/assets/20705647/46fc6d77-700b-4a18-918b-92e45359f4b0)

Works great:
![image](https://github.com/Skipperro/pajgps-homeassistant/assets/20705647/19a75da1-6118-497a-9016-ebee5b89232d)

If you want I could add coordinates as numbers to the list of entities, but could you explain me what is it you want to achieve by having them as numbers?

**Comment by klehnst** [comment]
Thank you for the information and sorry for your wasted time.
It was my mistake, I didn't know that the coordinates were there.

but how is it marked where you went? I did exactly as you did, I created a user as a vehicle and set the PAJ as a tracker.
Is this an additional integration?

**Comment by skipperro** [comment]
It's not wasted time. Now I know that it would probably make sense to add a more detailed info to the ReadMe file on how to use map card. Or a documentation link at least. 
https://www.home-assistant.io/dashboards/map/

Yes, it's a different integration, but the map card is a standard component in Home Assistant so you don't have to install anything else.

Here is a quick tutorial on how to replicate the map I have in my dashboard:

1. While on Dashboard, enable Dashboard Edit (°°° menu in top-right corner)
![image](https://github.com/Skipperro/pajgps-homeassistant/assets/20705647/0382d0d5-6506-4f56-97bb-a621b6402b6e)
2. Add a new card and set it to "Panel (1 card") mode:
![image](https://github.com/Skipperro/pajgps-homeassistant/assets/20705647/0015ca05-126c-473a-a5a3-07f2e28935ea)
3. Add a card of the "Map Card" type and configure it as you like. For me default zoom of 12 and tracking of last 5 hours works best. Don't forget to add entities to track - in your case that would be your PAJ GPS Tracker.
![image](https://github.com/Skipperro/pajgps-homeassistant/assets/20705647/b4cba949-038a-47ef-986e-5db3dbacda89)

Let me know if that works for you. :smile:

**Comment by klehnst** [comment]
Nice! It works! Thank you for your time.
Is it possible to change the timespan like the energy dashboard? That would be really cool.

I think that guide should be insert to the description. that will be good for newbees like me.

**Comment by skipperro** [comment]
@klehnst what do you mean by timespan?

- If you mean the time in which points on the map are displayed - that's the "Hours to show" in the map card configuration.

- If you mean update-time of the GPS - that's fixed at 30 seconds. I could expose settings to change that value, but only to increase it to something more than 30 seconds. If I'll request the data from PAJ GPS API more frequently I will be putting too much load on their servers.

**Comment by klehnst** [comment]
![image](https://github.com/Skipperro/pajgps-homeassistant/assets/46057062/e652e1a2-d7e1-4567-9a49-771ec52ac70b)

Yes i mean the time in which points on the map are displayed but like the energydashboard, in the energy dashboard you can select the date and times at the top, then you can see the energy production/consumption for the desired day/time.

It´s a example what i mean
![image](https://github.com/Skipperro/pajgps-homeassistant/assets/46057062/9bb5b66b-c501-4822-af77-4b3503f8617f)

**Comment by skipperro** [comment]
I don't think that standard map card supports this and history can give you history of speed and area you were located. I haven't saw the option to view points from specific period, but the raw data should be somewhere inside the HA, so.... maybe someone made some extended version of map card that is available as a plugin? 

If you'll find it, please let me know, I would also like to have such feature in my dashboard.

**Comment by klehnst** [comment]
I probably now have a reason to have the GPS data in a separate entity.
It is possible to use the GPS data for the weather in the vehicle.
I have not written this code myself, but it should work.

Have a look at it.


`type: custom:config-template-card
variables:
  LAT: states['sensor.gps_latitude'].state
  LON: states['sensor.gps_longitude'].state
  LAD: states['sensor.gps_latitude'].state
  LOM: states['sensor.gps_longitude'].state
entities:
  - sensor.gps_latitude
  - sensor.gps_longitude
card:
  type: iframe
  url: >-
    ${"https://radar.wo-cloud.com/mobile/embedded?wrx="+LAT+","+LON+"&wrm=8&wry="+LAD+","+LOM+"&center="+LAD+","+LOM}
  style:
    top: 100%
    left: 100%`

**Comment by DanielWeeber** [comment]
You can always access the data via the object.
just use something like "device_tracker.your_pajgps_object.latitude"

If you still need it directly in a sensor as state you can use a Template (https://www.home-assistant.io/integrations/template/) for that
